esbmc_ai/esbmc_util.py

```diff
@@ -94,10 +98,187 @@ def esbmc(path: str, esbmc_params: list, timeout: Optional[float] = None):
+    esbmc_output = re.sub(pattern, '', esbmc_output, flags=re.MULTILINE)
+    
+    esbmc_output = re.sub(r'\n\s*\n', '\n', esbmc_output)
+    esbmc_output = esbmc_output.replace("output from ESBMC", "ESBMC output")
+    esbmc_output = re.sub(r'The time.*', '', esbmc_output, flags=re.MULTILINE) 
+    
+    esbmc_output = re.sub(r'[^.]VCCs represent.*|[^.]These VCCs represent.*', '', esbmc_output, flags=re.MULTILINE) #These types of lines represent some sequences of VCCs that can easily be ommitted for development purposes
+    esbmc_output = re.sub(r'.*bitvector.*', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'\s*\(\s*`thread\d+`(?:,\s*`thread\d+`)*\s*\)', '', esbmc_output)
+    
+    esbmc_output = re.sub(r'.*floating-point arithmetic.*', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'.*encoding.*', '', esbmc_output, flags = re.MULTILINE)
+    esbmc_output = re.sub(r'.*encoding time.*', '', esbmc_output, flags= re.MULTILINE)
+
+    esbmc_output = re.sub(r', which takes \d+\.\d+s\.*', '.', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'The output indicates that ESBMC generates a verification condition \(VCC\) for each possible path and encodes them into a solver\.', '', esbmc_output, flags= re.MULTILINE) #This information can be disregarded for development purposes
+    esbmc_output = re.sub(r'Measures the [^.]+?encoding time [^.]+?\.', '', esbmc_output, flags=re.MULTILINE)
+
+    esbmc_output = re.sub(r'In this case, the procedure took [0-9.]+ seconds\.', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'\s*It measures[^.]+? encoding time [^.]+?\.', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'[^.]*GOTO program[^.]*\.', '', esbmc_output)
+
+    esbmc_output = re.sub(r'\d+\.\d+s \.', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'\.\s*.*time.*\n\n|\.\s*[^.]*time[^.]*\.', '', esbmc_output, flags=re.MULTILINE)    
+    esbmc_output =  re.sub(r'.*program time: \d+\.\d+s \.', '',esbmc_output, flags=re.MULTILINE)
+
+    esbmc_output = re.sub(r'\.\s*[^.]*?BMC program time:\d+\.\d+s', '', esbmc_output, flags= re.MULTILINE)
+    esbmc_output =re.sub(r'\.\s*[^.]*?Runtime decision procedure: \d+.\d+s\"*', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'.*Symex.*|.*symex.*|.*symbolic execution.*\n\n', '', esbmc_output, flags=re.MULTILINE)
+
+    esbmc_output =re.sub(r'After[^.]*?, the code', 'The program then', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'.*assignments.*|.*assignment.*', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'.*iteratively.*', '', esbmc_output, flags=re.MULTILINE)
+
+    esbmc_output = re.sub(r'.*branches.*', '', esbmc_output, flags=re.MULTILINE) #The informations about branches are not necessary relevant to the understanding of bugs
```

Comment:
Consider the following sample:

```c
#include <string.h>
#include <assert.h>

int main() {
	char* value = "branches";
	assert(strcmp("branches", value) == 0);
	return 0;
}
```

Would produce the following ESBMC output:

```
ESBMC version 7.5.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing sample.c
Converting
Generating GOTO Program
GOTO program creation time: 0.097s
GOTO program processing time: 0.002s
Starting Bounded Model Checking
Unwinding loop 8 iteration 1   file string.c line 96 column 12 function strcmp
Unwinding loop 8 iteration 2   file string.c line 96 column 12 function strcmp
Unwinding loop 8 iteration 3   file string.c line 96 column 12 function strcmp
Unwinding loop 8 iteration 4   file string.c line 96 column 12 function strcmp
Unwinding loop 8 iteration 5   file string.c line 96 column 12 function strcmp
Unwinding loop 8 iteration 6   file string.c line 96 column 12 function strcmp
Unwinding loop 8 iteration 7   file string.c line 96 column 12 function strcmp
Unwinding loop 8 iteration 8   file string.c line 96 column 12 function strcmp
Symex completed in: 0.002s (92 assignments)
Slicing time: 0.000s (removed 90 assignments)
Generated 19 VCC(s), 1 remaining after simplification (2 assignments)
No solver specified; defaulting to Boolector
Encoding remaining VCC(s) using bit-vector/floating-point arithmetic
Encoding to solver time: 0.000s
Solving with solver Boolector 3.2.3
Runtime decision procedure: 0.000s
Building error trace

[Counterexample]


State 1 file sample.c line 6 column 2 function main thread 0
----------------------------------------------------
Violated property:
  file sample.c line 6 column 2 function main
  assertion strcmp("branches", value) == 0
  return_value$_strcmp$1 == 0


VERIFICATION FAILED
```

Putting that through a regex viewer removes from the `Violated property:` section the line `assertion strcmp("branches", value) == 0`.

There are many more regex that are too generic to be included. You filter out ESBMC output that is useful.